specs/projectreuniondeployment/projectreuniondeployment.md

```diff
@@ -0,0 +1,138 @@
+
+Developers must not assume that the Project Reunion main package is present, or that the version
+present is the minimum version required to support the feature set they expect. Instead, this API
+should be used to validate any Project Reunion requirements and direct the user to resolve any
+missing dependencies.
+
+The developer scenarios for this API is to answer the questions:
+1. Does this system have the min version of Project Reunion I require?
+2. What do I do to resolve this if it doesn't?
+
+The API addresses #1 by a simple query of the version they seek and a possible check for OS
+updates. #2 has two possible answers. First is direct resolution by the developer by acquiring and
+installing the packages if possible. Some programs will not have permission or motivation to do
+this, so the second answer is providing a link up to the user where the user can take action to
+install the package. The API will address this scenario by providing an answer as to what is
+required, where to get it, and where to go for instruction.
+
+# Examples
+
+## GetProjectReunionStatus
+
+This is the means by which an app or program can call to verify Project Reunion is present to a
+minimum version. It should be called once after activation before trying to use Project Reunion
+functionality. This method returns a ProjectReunionStatus object that describes the current
+deployment state of Project Reunion.
+
+```C++
+Windows::ApplicationModel::PackageVersion version{1, 0, 0, 0};
+auto status =
+Microsoft::Management::Deployment::ProjectReunionDeployment::GetProjectReunionStatus(version);
+if (!status.IsOK())
+{
+    ...
+}
+```
+
+## GetProjectReunionPackageLink
+
+Provides a direct install link to deploy the latest version of ProjectReunion. If no OS Updates are
+required and the program or app has permissions to deploy packages, this can directly resolve the
+missing dependency without burdening the user.
+
+```C++
+Windows::Foundation::Uri packageLink = Microsoft::Management::Deployment::ProjectReunionDeployment::GetProjectReunionPackageLink();
+```
+
+## GetProjectReunionInstallInformationLink
+
+Provides a user-friendly link to acquire the latest version of Project Reunion and any OS updates
+that may be required. This can be used to guide the user to where they can resolve the missing
+dependency.
+
+```C++
+Windows::Foundation::Uri installInformationLink =
+Microsoft::Management::Deployment::ProjectReunionDeployment::GetProjectReunionInstallInformationLink();
+```
+
+# API Details
+
+```c#
+namespace Microsoft.Management.Deployment
+{
+    /// Represents the current Deployment status of Project Reunion
+    runtimeclass ProjectReunionDeploymentStatus
+    {
+        /// Returns True if Project Reunion is the desired Version and no updates or packages
+        /// are required.
+        Boolean IsOK { get; };
+
+        // Returns True if Windows Updates are required for Project Reunion.
+        Boolean WindowsUpdatesRequired { get; };
```

Comment:
NIT: IsWindowsUpdateRequired ?

Q to API Review. What's the common pattern to apply here?

Reply:
`IsWindowsUpdateRequired` or `RequiresWindowsUpdate` maybe.